src/order/order.service.ts

```diff
@@ -576,4 +576,16 @@ export class OrderService {
     }
     return isEmpty(targets) ? null : targets;
   }
+
+  public async getOrderProductsByMemberId(memberId: string, productType?: string): Promise<OrderProduct[]> {
+    const orderProducts = await this.entityManager.getRepository(OrderProduct).find({
+      where: {
+        order: {
+          memberId: memberId,
+        },
+        ...(productType && { product: { type: productType } }),
+      },
+    });
```

Comment:
Business logic is written in the service layer, and data queries are written in the infra layer. So it should be written in `order.infra.ts`.

Reply:
updated, thanks for reminder.